### PR TITLE
Updated test suite

### LINKED ISSUE #11: We need more unit tests!
@vivienr, @cdcapano, @cmbiwer, the unit tests currently only hit [22% of the code in `gwin` package](https://coveralls.io/builds/16385133) (this deliberately does not include the scripts under `/bin/`). I would like to get that number up around 90% (ideally, >75% more realistically) before anyone embarks on a grand new development project as part of `gwin`.

I know how to write unit tests, but have no idea how to actually test the code that exists in gwin at the moment, so am reaching out for help in writing new tests. The base plan is as follows:

- each `class` (e.g. `GwinClass`) has its own test class (`TestGwinClass`) (think of `unittest.TestCase`)
- each method of each `class` has its own unit test method as part of the test class
- each function has its own test function
- each line of code in each function should be tested somewhere, we can simplify annoying loops and the like by using [`fixtures`](https://docs.pytest.org/en/latest/fixture.html) and [parametrisation](https://docs.pytest.org/en/latest/parametrize.html).

Basically, every object we define in the package should have a corresponding test object. To that end I will set up a PR that defines a test class for every class definition I can find, and hope that y'all can help me populate it with useful stuff.

Testing the scripts in `/bin/` is another matter altogether, and not something I want to get to right now.

### LINKED ISSUE #16: Typo in InferenceFile.static_args
While writing unit tests, I discovered a typo in `InferenceFile.static_args`:

https://github.com/gwastro/gwin/blob/1c331ff65edc1dd94198967b6859abd96ddbf5a9/gwin/io/hdf.py#L167-L172

L172 should read `{arg: self.attrs[arg] ...`.

A fix will be included as part of #14.

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR updates the test suite to include a second test case module! and lots of other things:

- renamed `test/test_gwin.py` -> `test/test_sampler.py`
- added `test/test_io.py` (incomplete)
- added `test/test_likelihood.py` (incomplete)
- added `test/test_option_utils.py` (incomplete)
- added `test/test_results.py` (incomplete)
- added `test/utils` module with lots of fixtures
- added `conftest.py` with `--approximants` command-line options

The final point means that the suite can now be run using the default approximants with

```
python setup.py test  # OR
py.test
```

but you can specify which approximants to use:

```
py.test --approximants IMRPhenomA IMRPhenomB IMRPhenomC IMRPhenomD
```

Closes #11.
Fixes #16.

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
ok, @cmbiwer, I should probably stop here and add more unit tests in future PRs, you were automatically assigned because I modified `gwpy/io/hdf.py` to fix #16.

That commit (790ccfd) is the only to touch module code, all other modify build files, or test files.

File: conftest.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import pytest
+
+from pycbc.waveform import fd_approximants
+
+__author__ = 'Duncan Macleod <duncan.macleod@ligo.org>'
+
+DEFAULT_APPROXIMANTS = [
+    'TaylorF2',
+    'IMRPhenomPv2',
+]
```
Comment (vivienr) [inline conftest.py:32]: Could you add SEOBNRv4 to the default list? How long do the test take as is?

Comment (duncanmmacleod) [reply conftest.py:32]: @vivienr, on my macbook I get the following error: 

```python
ValueError: Approximant SEOBNRv4 not available
```

Is there something special that needs to be done to set up this approximant?

Comment (duncanmmacleod) [reply conftest.py:32]: This is probably a consequence of the test only being set up for frequency-domain waveforms, at the moment. Hopefully I can work with @cdcapano in a future PR to add support for TD waveform approximants.

Comment (vivienr) [reply conftest.py:32]: It's only available in the time domain, and the tests only do calls in the frequency domain. If that's too much troubles, let's look at implementing time domain versions in another patch.

**Review by cmbiwer — APPROVED** [review]
@duncanmmacleod I feel like some of the codeclimate thresholds are a bit low. Like this is considered a failure:
```
Function test_add_sampler_option_group has 26 lines of code (exceeds 25 allowed). Consider refactoring.
```

Maybe at some future point these could be relaxed a bit.

Otherwise, this PR looks fine.

**Comment by vivienr** [comment]
@cmbiwer: I agree. In particular, I'd up the lines of code allowed to 50, the cognitive complexity to 10, allow up to 8 arguments to functions, and 750 lines of codes per file.

**Pull request merged**